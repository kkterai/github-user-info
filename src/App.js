import React, { Component } from 'react';

import Data from './Data'

import './App.css';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
        githubUsername: "",
        data: null,
        user: "",
    };
}

componentDidMount() {
  fetch('https://api.github.com/users', {
    // headers: {
    //     Links: '<https://api.github.com/user/repos?page=3&per_page=100>; rel="next"'
    // }
  })
    .then(response => response.json())
    .then(data => this.setState({ data }));
}

handleSubmit = event => {
  event.preventDefault();
  let user = this.searchUsers(this.state.githubUsername)
  this.setState({ 
    githubUsername: "",
    user: user[0]
  })
}

handleUsernameChange = event => {
  this.setState({
    githubUsername: event.target.value
  })
}

searchUsers = username => {
  return this.state.data.filter(object => object.login === username);
}

  render() {
    return (
      <div className="App">
           <form onSubmit={event => this.handleSubmit(event)}>
              <h2>Enter a Github username:</h2>
              <input type="text" name="githubUsername" onChange={event => this.handleUsernameChange(event)} value={this.state.githubUsername} />
          </form>
          <Data userInfo={this.state.user}/>
      </div>
    );
  }
}

export default App;