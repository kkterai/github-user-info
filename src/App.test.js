import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-dom/extend-expect';
import App from './App';

afterEach(cleanup);

it("renders", ()=> {
    const { asFragment } = render(<App text="Enter a Github username:" />);
    expect(asFragment()).toMatchSnapshot();
});

it("has text in h2", () => {
    const { getByTestId, getByText } = render(<App text="Enter a Github username:"/>);
    expect(getByTestId('h2tag')).toHaveTextContent("Enter a Github username:");
})