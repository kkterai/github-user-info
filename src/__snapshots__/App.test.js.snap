// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <form>
      <h2
        class="test"
        data-testid="h2tag"
      >
        "Enter a Github username:"
      </h2>
      <input
        name="githubUsername"
        type="text"
        value=""
      />
    </form>
    <div>
      <br />
      <img />
      <h2>
        <a />
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
